### ISSUE TITLE
Fix deprecated code and improve existing warning filters

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#8296]
_Issue [JP-3546](https://jira.stsci.edu/browse/JP-3546) was created on JIRA by [Ned Molter](https://jira.stsci.edu/secure/ViewProfile.jspa?name=emolter):_

Replace code that causes DeprecationWarnings, and fix warnings.simplefilter() all instances to filter only specific warnings on specific lines of code.  Some details below, more in comments as discovered:
 * A numpy array operation in wfs_combine is deprecated as of numpy 1.25. Traceback says 

```java
/Users/emolter/repositories/jwst/jwst/wfs_combine/wfs_combine.py:514: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)    sci_data[y, x] = griddata(ga[:, 0:2], ga[:, 2], [(y, x)], method='linear') ```
 * The same deprecation warning is being tripped inside astropy/io/fits/fitsrec.py when running regtest/test_nirspec_mos_spec3.py.  Inspection of the astropy code suggests this is not a problem with astropy, but instead with inappropriate input (function expecting int but getting a one-element array)
 * A [search for the keyword "simplefilter"](https://github.com/search?q=repo%3Aspacetelescope%2Fjwst+simplefilter&type=code) turned up a few instances of simplefilter("ignore") without a with block or any specificity

**Comment by stscijgbot-jp** [comment]
_Comment by [Ned Molter](https://jira.stsci.edu/secure/ViewProfile.jspa?name=emolter) on [JIRA](https://jira.stsci.edu/browse/JP-3546?focusedCommentId=685755#comment-685755):_

Three deprecated model types are being tested in test_model_containers.py::test_all_datamodels_init; see messages below.  The argument could be made, though, that these test cases should not be removed until the removal is enforced.  Planning to leave these as-is unless this deprecation will be enforced in 10.2

 
jwst/datamodels/tests/test_model_container.py::test_all_datamodels_init[DrizProductModel]
  /data1/jenkins/workspace/RT/JWST-Developers-Pull-Requests/miniconda/envs/tmp_env2/lib/python3.11/site-packages/stdatamodels/jwst/datamodels/drizproduct.py:10: DeprecationWarning: DrizProduct is deprecated and will be removed.  Use ImageModel.
    warnings.warn(message="DrizProduct is deprecated and will be removed.  "

jwst/datamodels/tests/test_model_container.py::test_all_datamodels_init[MultiProductModel]
  /data1/jenkins/workspace/RT/JWST-Developers-Pull-Requests/miniconda/envs/tmp_env2/lib/python3.11/site-packages/stdatamodels/jwst/datamodels/multiprod.py:11: DeprecationWarning: MultiProductModel is deprecated and will be removed.  Use MultiSlitModel.
    warnings.warn(message="MultiProductModel is deprecated and will be removed.  "

jwst/datamodels/tests/test_model_container.py::test_all_datamodels_init[MIRIRampModel]
  /data1/jenkins/workspace/RT/JWST-Developers-Pull-Requests/miniconda/envs/tmp_env2/lib/python3.11/site-packages/stdatamodels/jwst/library/basic_utils.py:23: DeprecationWarning: "MIRIRampModel" is deprecated and will be removed. Use RampModel

**Comment by stscijgbot-jp** [comment]
_Comment by [Ned Molter](https://jira.stsci.edu/secure/ViewProfile.jspa?name=emolter) on [JIRA](https://jira.stsci.edu/browse/JP-3546?focusedCommentId=685756#comment-685756):_

use of stpipe.extern.configobj should be changed:

 
jwst/resample/resample_step.py:7
  /data1/jenkins/workspace/RT/JWST-Developers-Pull-Requests/clone/jwst/resample/resample_step.py:7: DeprecationWarning: stpipe.extern.configobj is deprecated in favor of astropy.extern.configobj, please use that instead.
    from stpipe.extern.configobj.validate import Validator

**Comment by stscijgbot-jp** [comment]
_Comment by [Ned Molter](https://jira.stsci.edu/secure/ViewProfile.jspa?name=emolter) on [JIRA](https://jira.stsci.edu/browse/JP-3546?focusedCommentId=685757#comment-685757):_

two instances of scipy.integrate.simps inside ami/instrument_data.py should be changed to scipy.integrate.simpson

 
jwst/regtest/test_niriss_ami3.py::test_ami_analyze_with_nans
  /data1/jenkins/workspace/RT/JWST-Developers-Pull-Requests/clone/jwst/ami/instrument_data.py:83: DeprecationWarning: 'scipy.integrate.simps' is deprecated in favour of 'scipy.integrate.simpson' and will be removed in SciPy 1.14.0
    area = simps(thru_st_0, thru_st_1)

jwst/regtest/test_niriss_ami3.py::test_ami_analyze_with_nans /data1/jenkins/workspace/RT/JWST-Developers-Pull-Requests/clone/jwst/ami/instrument_data.py:94: DeprecationWarning: 'scipy.integrate.simps' is deprecated in favour of 'scipy.integrate.simpson' and will be removed in SciPy 1.14.0 area = simps(wt, wl)

**Comment by braingram** [comment]
I would vote for deleting test_all_datamodels_init. It is effectively a duplicate of a test in stdatamodels:
https://github.com/spacetelescope/stdatamodels/blob/e91f81f037029a7d58ee8f822cef61ad1d789de9/src/stdatamodels/jwst/datamodels/tests/test_models.py#L313
It is parametrized with the same values (which do not include `SourceModelContainer` so the conditional that differs in the jwst test is never visited). It is likely this test could have been removed in https://github.com/spacetelescope/jwst/pull/7439

**Comment by stscijgbot-jp** [comment]
_Comment by [Ned Molter](https://jira.stsci.edu/secure/ViewProfile.jspa?name=emolter) on [JIRA](https://jira.stsci.edu/browse/JP-3546?focusedCommentId=686116#comment-686116):_

Regression tests with "-W error" turned up many instances of a warning within ASDF that reads
```java
asdf.exceptions.AsdfConversionWarning: A ndarray subclass (<class 'astropy.io.fits.fitsrec.FITS_rec'>) was converted as a ndarray. This behavior will be removed from a future version of ASDF. See https://asdf.readthedocs.io/en/latest/asdf/config.html#convert-unknown-ndarray-subclasses" ```
This isn't specifically labeled a deprecation warning, but based on the message, that's what it is in practice.  The full traceback is found e.g. [here](https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1235/testReport/junit/jwst.regtest/test_nirspec_fs_spec2/*stable_deps__test_nirspec_fs_spec2_jw02072_o002_20221206t143745_spec2_00001_asn_json_nsclean*/).  I assume this should be handled/fixed inside stdatamodels, and not in the jwst pipeline, e.g. at this step of the traceback:
```java
File "/data1/jenkins/workspace/RT/JWST-Developers-Pull-Requests/miniconda/envs/tmp_env2/lib/python3.11/site-packages/stdatamodels/fits_support.py", line 507, in _create_asdf_hdu ```
I'm still pretty unfamiliar with asdf and stdatamodels though, so I'm not sure I'm interpreting this correctly.  Do people have opinions on what should be done here?  Should a separate ticket be filed that asks for this to be handled in stdatamodels?  Should I just suppress this warning in the jwst pipeline?

**Comment by stscijgbot-jp** [comment]
_Comment by [Ned Molter](https://jira.stsci.edu/secure/ViewProfile.jspa?name=emolter) on [JIRA](https://jira.stsci.edu/browse/JP-3546?focusedCommentId=686311#comment-686311):_

Running the test suite turning warnings into errors revealed a bunch of warnings that fall outside the scope of this ticket, and which might take some investigation to ensure proper resolution (rather than just suppressing them).  Most of these are assumed to be benign, but I thought it would be worth listing them here for reference.
 * jwst/associations/association.py:215: UserWarning: 'expname' contains path, but should only be a filename.  All input files should be in a single directory, so no path is needed. seen in e.g. **jwst.skymatch.tests.test_skymatch.[stable-deps] test_asn_input**
 * /data1/jenkins/workspace/RT/JWST-Developers-Pull-Requests/miniconda/envs/tmp_env2/lib/python3.11/site-packages/stdatamodels/validate.py:38: ValidationWarning. While validating meta.exposure.readpatt the following error occurred: Failed validating 'enum' in schema. seen in e.g. *jwst.regtest.test_nirspec_verify.[stable-deps] test_verify_detector1[dq_init]*
 * photutils.utils.exceptions.NoDetectionsWarning: No sources were found. /data1/jenkins/workspace/RT/JWST-Developers-Pull-Requests/clone/jwst/tweakreg/tweakreg_catalog.py:144: in _DaoStarFinderWrapper. sources = starfind(data, mask=mask). seen in e.g. *jwst.regtest.test_nircam_image.[stable-deps] test_image3_closedfile.*
 * TypeError: expected str, bytes or os.PathLike object, not NoneType. okify_schema = generate_upload_schema(okify_schema_pattern, f"\{os.path.dirname(rtdata.truth_remote)}/“). inside pytest function for *jwst.regtest.test_nircam_mtimage.[stable-deps] test_nircam_image_moving_target_kwds[midpt_not_in_mt_table_range]*
 * /data1/jenkins/workspace/RT/JWST-Developers-Pull-Requests@2/miniconda/envs/tmp_env2/lib/python3.11/site-packages/stdatamodels/jwst/datamodels/util.py:242: NoTypeWarning. seen in e.g. *jwst.regtest.test_fgs_guider.[stable-deps] test_fgs_guider[exptype_fgs_acq1-cal]* 
 * numpy.RankWarning: Polyfit may be poorly conditioned. jwst/emicorr/emicorr.py:455: in apply_emicorr: m, b = np.polyfit(lut_reference, lut, 1). seen in **jwst.emicorr.tests.test_emicorr.[stable-deps] test_do_correction**
 * a bunch of runtime warnings in Numpy for things like taking means of empty slice or dividing by zero

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-3546?focusedCommentId=690251#comment-690251):_

Fixed by #8320
